configure.ac

```diff
@@ -727,6 +727,10 @@ fi
 
 AC_CHECK_HEADERS([endian.h sys/endian.h byteswap.h stdio.h stdlib.h unistd.h strings.h sys/types.h sys/stat.h sys/select.h sys/prctl.h])
 
+AC_CHECK_DECLS([getifaddrs, freeifaddrs],,,
+    [#include <sys/types.h>
+    #include <ifaddrs.h>]
+)
```

Comment:
Seems a bit simpler overall to add ifaddrs.h to `AC_CHECK_HEADERS` above, as it seems android does not include the header at all, and switch on `HAVE_SYS_TYPES_H` and `HAVE_IFADDERS_H`